contact/toolsPane.js

```diff
@@ -164,46 +162,47 @@ var toolsPane = function (selectAllGroups, selectedGroups, groupsMainTable, book
                 }
               })
             }
-          })
-
-          var filesToDelete = [ card.doc() ]
-          var photo = kb.any(card, ns.vcard('hasPhoto'))
-          if (photo) filesToDelete.push(photo)
-          filesToDelete.push(card.dir()) // the folder last
-          log('Files to delete: ' + filesToDelete)
-          if (!confirm('DELETE card ' + card.dir() + ' for ' + name + ', with ' + desc.length + 'statements?')){
-            return resolve('Cancelled by user')
-          }
+            var filesToDelete = [ card.doc() ]
+            var photos = kb.each(card, ns.vcard('hasPhoto')) // could be > 1
+            if (photos.length) {
+              filesToDelete = filesToDelete.concat(photos)
+            }
+            filesToDelete.push(card.dir()) // the folder last
+            log('Files to delete: ' + filesToDelete)
+            if (!confirm('DELETE card ' + card.dir() + ' for "' + kb.any(card, VCARD('fn')) + '", with ' + kb.each(card).length + 'statements?')){
+              return resolve('Cancelled by user')
+            }
 
-          var deleteOneFile = function() {
-            var resource = filesToDelete.shift()
-            if (!resource) {
-              log('All deleted')
-              removeFromMainIndex()
-              return
+            var deleteNextFile = function () {
```

Comment:
Nit, but `function deleteNextFile ()` for such situations might be more idiomatic.

Reply:
They are different, aren't they?   function x()  is hoisted up to th top of the containing function.  I thought.

Reply:
Yes, but since you're not referring to `deleteNextFile` above this line, they're the same.

Reply:
OK